src/components/audio-zone.js

```diff
@@ -42,17 +49,35 @@ AFRAME.registerComponent("audio-zone", {
     this.enableDebug(this.data.debuggable && window.APP.store.state.preferences.showAudioDebugPanel);
   },
 
+  tick: (() => {
+    const quat = new THREE.Quaternion();
+    return function() {
+      this.el.object3D.quaternion.set(0, 0, 0, 1);
+      this.el.object3D.quaternion.multiply(this.el.object3D.parent.getWorldQuaternion(quat).inverse());
+      this.el.object3D.updateMatrix();
+    };
+  })(),
+
   isEnabled() {
     return this.data.enabled;
   },
 
   getBoundingBox: (function() {
     const bbaa = new THREE.Box3();
     return function() {
+      this.debugMesh.geometry.computeBoundingBox();
       return bbaa.copy(this.debugMesh.geometry.boundingBox).applyMatrix4(this.el.object3D.matrixWorld);
     };
   })(),
 
+  getBoundingSphere: (function() {
+    const bbaa = new THREE.Sphere();
+    return function() {
+      this.debugMesh.geometry.computeBoundingSphere();
```

Comment:
Better to add a guard for efficiency unless this method is guaranteed to be called only once or the bounding sphere really needs to be recomputed for each call?

```
if (this.debugMesh.geometry.boundingSphere !== null) {
  this.debugMesh.geometry.computeBoundingSphere();
}
```

Reply:
Oh yep, there's no need at all. Updated.